sia/wallet/outputs.go

```diff
@@ -63,7 +63,12 @@ func (w *Wallet) findOutputs(amount consensus.Currency) (spendableOutputs []*spe
 
 	// This code will only be reached if total < amount, meaning insufficient
 	// funds.
-	err = fmt.Errorf("insufficient funds, requested %v but only have %v", amount, total)
+	//
+	// TODO: Figure out some way to return a named error that also contains
+	// custom information.
+	// err = fmt.Errorf("insufficient funds, requested %v but only have %v", amount, total)
+	err = components.LowBalanceErr
+
```

Comment:
stdlib would do this with a new error type and a type assertion. See `os.IsExist` for an equivalent.
```go
// in components package(?)
type balanceErr struct {
    error
}

func IsBalanceErr(err error) bool {
    _, ok := err.(*balanceErr)
    return ok
}

// in findOutputs:
err = balanceErr(amount, total)

// in proposeSmallContract:
if err != nil && components.IsBalanceErr(err) {
    ... 
}
```